extensions/src/main/java/io/lionweb/serialization/extensions/ExtendedLionWebRepoClient.java

```diff
@@ -0,0 +1,231 @@
+package io.lionweb.serialization.extensions;
+
+import com.google.gson.*;
+import io.lionweb.lioncore.java.LionWebVersion;
+import io.lionweb.lioncore.java.model.ClassifierInstance;
+import io.lionweb.lioncore.protobuf.PBBulkImport;
+import io.lionweb.repoclient.LionWebRepoClient;
+import io.lionweb.repoclient.RequestFailureException;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import okhttp3.*;
+
+public class ExtendedLionWebRepoClient extends LionWebRepoClient implements AdditionalAPIClient {
+
+  private static final MediaType PROTOBUF = MediaType.get("application/protobuf");
+  private static final MediaType FLATBUFFERS = MediaType.get("application/flatbuffers");
+
+  public class Builder extends LionWebRepoClient.Builder {
+    @Override
+    public ExtendedLionWebRepoClient build() {
+      return new ExtendedLionWebRepoClient(
+          lionWebVersion,
+          hostname,
+          port,
+          authorizationToken,
+          clientID,
+          repository,
+          connectTimeoutInSeconds,
+          callTimeoutInSeconds);
+    }
+  }
+
+  public ExtendedLionWebRepoClient(
+      LionWebVersion lionWebVersion, String hostname, int port, String repository) {
+    super(lionWebVersion, hostname, port, repository);
+  }
+
+  public ExtendedLionWebRepoClient(
+      LionWebVersion lionWebVersion,
+      String hostname,
+      int port,
+      String authorizationToken,
+      String clientID,
+      String repository,
+      long connectTimeoutInSeconds,
+      long callTimeoutInSeconds) {
+    super(
+        lionWebVersion,
+        hostname,
+        port,
+        authorizationToken,
+        clientID,
+        repository,
+        connectTimeoutInSeconds,
+        callTimeoutInSeconds);
+  }
+
+  @Override
+  public void bulkImport(
+      @Nonnull BulkImport bulkImport, TransferFormat transferFormat, Compression compression)
+      throws IOException {
+    if (bulkImport.isEmpty()) {
+      return;
+    }
+    switch (transferFormat) {
+      case JSON:
+        bulkImportUsingJson(bulkImport, compression);
+        return;
+      case PROTOBUF:
+        bulkImportUsingProtobuf(bulkImport, compression);
+        return;
+      case FLATBUFFERS:
+        bulkImportUsingFlatbuffers(bulkImport, compression);
+        return;
+      default:
+        throw new UnsupportedOperationException();
+    }
+  }
+
+  @Override
+  public List<NodeInfo> getNodeTree(List<String> nodeIDs, @Nullable Integer depthLimit)
+      throws IOException {
+    if (nodeIDs.isEmpty()) {
+      return Collections.emptyList();
+    }
+    String url = protocol.value + "://" + hostname + ":" + port + "/additional/getNodeTree";
```

Comment:
This feels like an invitation for security issues. Isn't there a safer API on e.g. `HttpUrl`?